torch/autograd/gradcheck.py

```diff
@@ -72,6 +72,55 @@ def _iter_tensors(x: Union[torch.Tensor, Iterable[torch.Tensor]],
                 yield result
 
 
+def _densify(x):
+    # return a copy of sparse x with all unspecified elements
+    # "replaced" with zero-valued elements
+    if isinstance(x, (list, tuple)):
+        return type(x)(map(_densify, x))
+    elif not is_tensor_like(x) or x.layout in {torch.strided, torch._mkldnn}:  # type: ignore[attr-defined] # no attr _mkldnn
+        return x
+    elif x.layout is torch.sparse_coo:
+        def get_stride(size):
```

Comment:
You don't need this logic.
Just read the stride from the result of `torch.ones()` before doing `to_sparse()` on it.

Reply:
Right, thanks for the hint! Fixed.